azure/imageclient/client.go

```diff
@@ -0,0 +1,62 @@
+package imageclient
+
+import (
+	"encoding/xml"
+	"errors"
+	"fmt"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/azure"
+)
+
+const (
+	azureImageListURL    = "services/images"
+	errInvalidImage      = "Can not find image %s in specified subscription, please specify another image name."
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+//ImageClient is used to manage operations on Azure Locations
+type ImageClient struct {
```

Comment:
Moving the struct is fine, I'd keep the `NewClient()` in the main client file in each case though as it makes it more discoverable when browsing the code.